### ISSUE TITLE
No pdf output

### ISSUE THREAD

**Issue by Justinzobel** [adamrehn/gen-invoice#1]
I'm getting the .html output but not pdf

```
~ > gen-invoice 1707 ffc ffc-04.csv ffc-2020-04
Loading data from the following files:
Items:      /home/justin/ffc-04.csv
Payee:      /home/justin/.config/gen-invoice/payees/1707.yml
Payer:      /home/justin/.config/gen-invoice/payers/ffc.yml
Stylesheet: /home/justin/.config/gen-invoice/styles/default.css
Template:   /home/justin/.config/gen-invoice/templates/default.html

Error: module 'humanfriendly' has no attribute 'prompts'
```

Installed via:
```
sudo pip3 install gen-invoice
npm install electron-pdf (as it said it needed it)
```

**Comment by Justinzobel** [comment]
It appears the above error only happens once I initially run it and it complains about the electron-pdf missing.

I've even tried installing electron-pdf as root as described on their page:
```
sudo npm install electron-pdf -g --unsafe-perm
/usr/bin/electron-pdf -> /usr/lib/node_modules/electron-pdf/cli.js

> core-js@3.6.5 postinstall /usr/lib/node_modules/electron-pdf/node_modules/core-js
> node -e "try{require('./postinstall')}catch(e){}"

Thank you for using core-js ( https://github.com/zloirock/core-js ) for polyfilling JavaScript standard library!

The project needs your help! Please consider supporting of core-js on Open Collective or Patreon: 
> https://opencollective.com/core-js 
> https://www.patreon.com/zloirock 

Also, the author of core-js ( https://github.com/zloirock ) is looking for a good job -)


> electron@7.2.4 postinstall /usr/lib/node_modules/electron-pdf/node_modules/electron
> node install.js

+ electron-pdf@7.0.0
added 138 packages from 403 contributors in 37.849s
```

**Comment by adamrehn** [comment]
It's quite possible that electron-pdf has changed since the version that gen-invoice was designed to work with. Does uninstalling the current version and then installing version 4.0.6 using `npm install -g electron-pdf@4.0.6` solve the issue?

**Comment by Justinzobel** [comment]
Thanks for the quick response. I removed electron-pdf then tried installing as suggested but it errors out:

```
~ > sudo npm install -g electron-pdf@4.0.6
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
/usr/bin/electron-pdf -> /usr/lib/node_modules/electron-pdf/cli.js

> electron@4.1.4 postinstall /usr/lib/node_modules/electron-pdf/node_modules/electron
> node install.js

Downloading tmp-38156-0-electron-v4.1.4-linux-x64.zip
Error: EACCES: permission denied, open '/root/.cache/electron/tmp-38156-0-electron-v4.1.4-linux-x64.zip'
/usr/lib64/node_modules/electron-pdf/node_modules/electron/install.js:49
  throw err
  ^

[Error: EACCES: permission denied, open '/root/.cache/electron/tmp-38156-0-electron-v4.1.4-linux-x64.zip'] {
  errno: -13,
  code: 'EACCES',
  syscall: 'open',
  path: '/root/.cache/electron/tmp-38156-0-electron-v4.1.4-linux-x64.zip'
}
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! electron@4.1.4 postinstall: `node install.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the electron@4.1.4 postinstall script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2020-05-01T05_33_55_609Z-debug.log
```

**Comment by Justinzobel** [comment]
Content of the log file (last line of previous comment) https://hastebin.com/alufuvigod

**Comment by adamrehn** [comment]
A quick Google suggests that the error you saw there is a common issue: https://docs.npmjs.com/resolving-eacces-permissions-errors-when-installing-packages-globally

**Comment by Justinzobel** [comment]
I tried that but got no further. I'll wait for a version that supports the current electron-pdf if that's going to happen.

**Comment by Justinzobel** [comment]
Is an update of gen-invoice likely to happen to align it with current dependency versions?

**Comment by adamrehn** [comment]
Yes, I'll certainly be updating gen-invoice to address this, most likely incorporating support for [npx](https://www.npmjs.com/package/npx) to avoid the installation issues you encountered. I'll update this issue once the fix is implemented.

**Comment by Justinzobel** [comment]
Excellent, many thanks!

**Comment by Justinzobel** [comment]
In the interim if anyone is having issues with this I have opened the html output in Chrome and used Print to PDF as a workaround.

**Comment by Justinzobel** [comment]
@adamrehn Any progress on this?

**Comment by adamrehn** [comment]
@Justinzobel no, but it's in my backlog of assigned issues, so it won't get forgotten.

**Comment by Justinzobel** [comment]
Have confirmed this solves the issue.

**Comment by adamrehn** [comment]
I've conditionally applied the `shell=False` fix from #4 under non-Windows platforms in commit 72c2580 (since it breaks things under Windows) and switched to using `npx` to invoke electron-pdf in commit 9cc450e. This should fix this issue and also provide better robustness when an incompatible version of electron-pdf is installed. The fixes are now live in gen-invoice version 0.0.2.

**Comment by Justinzobel** [comment]
Thanks!
